File: backend/models/Doctor.js
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  name: { type: String, required: true },
  email: { type: String, required: true },
  specialization: { type: String },
}, { timestamps: true });

module.exports = mongoose.model('Doctor', doctorSchema);

File: backend/models/doctor.py
from mongoengine import Document, StringField, ReferenceField, DateTimeField
from models.user import User
from datetime import datetime

class Doctor(Document):
    userId = ReferenceField(User, required=True, reverse_delete_rule=2)  # CASCADE delete
    name = StringField(required=True)
    email = StringField(required=True)
    specialization = StringField()
    
    created_at = DateTimeField(default=datetime.utcnow)
    updated_at = DateTimeField(default=datetime.utcnow)

    meta = {
        'collection': 'doctor',
        'indexes': ['email'],
        'ordering': ['-created_at'],
        'auto_create_index': True
    }

    def save(self, *args, **kwargs):
        self.updated_at = datetime.utcnow()
        return super(Doctor, self).save(*args, **kwargs)
